analyses/gistic-cohort-vs-histology-comparison/gistic-cohort-vs-histology-comparison.Rmd

```diff
@@ -0,0 +1,342 @@
+medulloblastoma_gistic_zip <- file.path(data_dir, "pbta-cnv-consensus-medulloblastoma-gistic.zip")
+
+if (!dir.exists(medulloblastoma_gistic_dir)) {
+  unzip(medulloblastoma_gistic_zip, 
+        exdir = data_dir)
+}
+
+```
+
+### Run `plot_gistic_scores` for each file
+
+```{r}
+# Define file paths to `scores.gistic` files and run `plot_gistic_scores`
+# function for each file
+cohort_scores <-
+  file.path(cohort_gistic_dir, "scores.gistic")
+cohort_scores_plot <- plot_gistic_scores(cohort_scores)
+
+lgat_scores <-
+  file.path(lgat_gistic_dir, "scores.gistic")
+lgat_scores_plot <- plot_gistic_scores(lgat_scores)
+
+hgat_scores <-
+  file.path(hgat_gistic_dir, "scores.gistic")
+hgat_scores_plot <- plot_gistic_scores(hgat_scores)
+
+medulloblastoma_scores <-
+  file.path(medulloblastoma_gistic_dir, "scores.gistic")
+medulloblastoma_scores_plot <-
+  plot_gistic_scores(medulloblastoma_scores)
+```
+
+### Generate multipanel plots 
+
+```{r fig.height = 10, fig.width = 24}
+lgat_plot_list <- list(lgat_scores_plot, cohort_scores_plot)
+generate_multipanel_plot(plot_list = lgat_plot_list,
+                         plot_title = "LGAT scores versus entire cohort",
+                         output_directory = plots_dir,
+                         output_filename = "lgat_gistic_scores_multipanel_plot.png")
+
+hgat_plot_list <- list(hgat_scores_plot, cohort_scores_plot)
+generate_multipanel_plot(plot_list = hgat_plot_list,
+                         plot_title = "HGAT scores versus entire cohort",
+                         output_directory = plots_dir,
+                         output_filename = "hgat_gistic_scores_multipanel_plot.png")
+
+medulloblastoma_plot_list <- list(medulloblastoma_scores_plot, cohort_scores_plot)
+generate_multipanel_plot(plot_list = medulloblastoma_plot_list,
+                         plot_title = "Medulloblastoma scores versus entire cohort",
+                         output_directory = plots_dir,
+                         output_filename = "medulloblastoma_gistic_scores_multipanel_plot.png")
+```
+
+## Compare `amp_genes.conf_90.txt` and `del_genes.conf_90.txt` files
+
+### Custom function
+```{r}
+plot_amp_del_genes <-
+  function(cohort_genes_file,
+           histology_genes_file) {
+    # Given the GISTIC result `amp_genes_conf_90.txt` or `del_genes_conf_90.txt`
+    # files for the entire cohort and a specific histology, plot a Venn Diagram
+    # showing the counts of rows that overlap/do not overlap between the two
+    # files.
+    #
+    # Args:
+    #   cohort_genes_file: the file path to the amp/del genes file for
+    #                           the entire cohort
+    #   histology_genes_file: the file path to the amp/del genes file
+    #                              for an individual histology
+    #
+    # Returns:
+    #  This function displays a Venn Diagram that represents the data that
+    #  overlaps/does not overlap between the two given files
+    
+    # Read in files
```

Comment:
You are treating the two files in the exact same way, which is a good indication to me that this can be wrapped up in a function. I can imagine a situation where you might want to examine what genes are in the vectors that underlie the Venn diagram, so you can separate things out into two function: 1) read in the file and process the vector and 2) create the Venn diagram.

With the first function, you'll only need to process the entire cohort results _once_.